hydromt/models/api.py

```diff
@@ -65,17 +64,17 @@ class Model(object, metaclass=ABCMeta):
         "tables": Dict[str, pd.DataFrame],
         "maps": XArrayDict,
         "forcing": XArrayDict,
-        "region": gpd.GeoDataFrame,
+        "region": ModelRegion,
         "results": XArrayDict,
         "states": XArrayDict,
     }
 
     def __init__(
         self,
         root: Optional[str] = None,
-        mode: Optional[str] = "w",
+        mode: str = "w",
```

Comment:
because we set the default to "w" so this is always a string. The only way this can be `None` is if a user explicitly sets `Model(model=None)` which doesn't make sense